packages/plugin-react/src/jsx-runtime/babel-restore-jsx.spec.ts

```diff
@@ -108,4 +108,10 @@ describe('babel-restore-jsx', () => {
       )
     ).toMatchInlineSnapshot(`"<h1>{foo ? <p /> : null}</h1>;"`)
   })
+
+  it('should handle lowercase component names', () => {
+    expect(jsx('React.createElement(aaa)')).toMatchInlineSnapshot(
+      `"React.createElement(aaa);"`
```

Comment:
I'm not 100% sure if leaving it untouched will work. Have you tested this in a real scenario?

Another option is to not use libraries that minify themselves ;P

Reply:
> I'm not 100% sure if leaving it untouched will work. Have you tested this in a real scenario?

I tried in dev and build, in SSR and in the browser on a mid-sized project, works for me™ :)

In any case, I would consider it a bug if it didn't. It's not illegal to mix the new JSX runtime and `React.createElement`. In fact, `createElement` is still the recommended way in some cases. [The blog post](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) that introduced the new runtime specifically says:

> If you need to manually create elements in your code, you should keep using React.createElement.

So things like `React.createElement(...args)` should (and does) keep working.

> Another option is to not use libraries that minify themselves ;P

Fair, but they're still widespread in the ecosystem :)